codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -409,15 +409,23 @@ let createBatch = (self: t, ~maxBatchSize: int, ~onlyBelowReorgThreshold: bool)
     ~onlyBelowReorgThreshold,
   )
 
+  // Needed to recalculate the computed queue sizes
+  let fetchStatesMap = fetchStatesMap->ChainMap.map(v => {
+    {
+      ...v,
+      partitionedFetchState: v.partitionedFetchState->PartitionedFetchState.syncStateOnQueueUpdate,
+    }
+  })
+
```

Comment:
Maybe this should be done inside getFetchStateWithData function on the shouldDeepCopy flag or with a different flag.

Reply:
During `shouldDeepCopy`, we don't know the number of events taken from the queue to batch, so we can't correctly update the state. I was actually thinking of changing the way how the batch is created so the part of the code is more reliable, but I scoped it out since this was not mandatory for the code to work.